apps/gallery/js/nfc.js

```diff
@@ -10,26 +10,75 @@
  * When the UI changes so that no sharable object is visible, call unshare().
  */
 var NFC = {
-  share: function share(f) {
-    // Enable NFC sharing by defining a "peerready" callback that
-    // calls sendFile() to transfer the current image or video to the peer
+  shrinkingUI: null,
+  peer: null,
+  share: function share(f, options) {
     if (navigator.mozNfc) {
-      navigator.mozNfc.onpeerready = function(event) {
-        if (typeof f === 'function') {
-          var promise = f();
-          promise.then(function(file) { event.peer.sendFile(file); });
-        }
-        else {
-          event.peer.sendFile(f);
+      // This is a workaround for bug 1163957. The complete solution should be
+      // done in bug 1109495.
```

Comment:
Please add to this comment to say that this workaround only applies to the open activity.